kdl_parser/CMakeLists.txt

```diff
@@ -2,43 +2,74 @@ cmake_minimum_required(VERSION 2.8.3)
 
 project(kdl_parser)
 
-find_package(Boost REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
-
-find_package(catkin REQUIRED
-  COMPONENTS rosconsole urdf cmake_modules
+find_package(catkin QUIET
+  COMPONENTS rosconsole cmake_modules
 )
+
+if(NOT catkin_FOUND)
+  list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+endif()
+
+find_package(urdfdom REQUIRED)
+include_directories(${urdfdom_INCLUDE_DIRS})
+
 find_package(orocos_kdl REQUIRED)
 find_package(TinyXML REQUIRED)
 find_package(TinyXML2 REQUIRED)
 
-include_directories(include ${orocos_kdl_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIRS} ${TinyXML2_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS})
+# check for rosconsole
+# We check additionally for catkin to distinguish between an "official" ROS distribution
+# and the one provided in the distribution's repository.
+find_package(rosconsole QUIET)
+if(rosconsole_FOUND AND catkin_FOUND)
+  add_definitions(-DHAS_ROS)
+endif()
+
+find_package(urdf QUIET)
+if(urdf_FOUND)
+  add_definitions(-DHAS_URDF)
+  include_directories(${urdf_INCLUDE_DIRS})
+endif()
 
-link_directories(${catkin_LIBRARY_DIRS})
-link_directories(${Boost_LIBRARY_DIRS})
+include_directories(include ${orocos_kdl_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIRS} ${TinyXML2_INCLUDE_DIRS})
 
 add_compile_options(-std=c++11)
 
-catkin_package(
-  LIBRARIES ${PROJECT_NAME} ${orocos_kdl_LIBRARIES}
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS rosconsole urdf
-  DEPENDS orocos_kdl TinyXML TinyXML2
-)
+if(catkin_FOUND)
+  link_directories(${catkin_LIBRARY_DIRS})
+  include_directories(${catkin_INCLUDE_DIRS})
+
+  catkin_package(
+    LIBRARIES ${PROJECT_NAME} ${orocos_kdl_LIBRARIES}
+    INCLUDE_DIRS include
+    CATKIN_DEPENDS rosconsole urdf
+    DEPENDS orocos_kdl TinyXML TinyXML2
+  )
+endif()
 
 add_library(${PROJECT_NAME} src/kdl_parser.cpp)
```

Comment:
It looks like there's some catkin magic here where when I build with `catkin_make` it creates a shared library, but when I build without ROS on ubuntu 18.04 it builds a static library. Maybe add `SHARED` to this call?

note to self for non-ros building
```
CMAKE_PREFIX_PATH=/opt/ros/melodic/share/orocos_kdl/cmake/ cmake ../src/kdl_parser/kdl_parser
```

Reply:
Good catch. It's shared now.